incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -97,3 +103,24 @@ func getSelectorFromString(str string) (labels.Selector, error) {
 	}
 	return selector, nil
 }
+
+// GetNoneSelector returns true indicates that user do not want this object to be propagated
+func GetNoneSelector(inst *unstructured.Unstructured) (bool, error) {
+	noneSelectorStr := GetNoneSelectorAnnotation(inst)
+	// Empty string is treated as 'false'. In other selector cases, the empty string is auto converted to
+	// a selector that matches everything.
+	if noneSelectorStr == "" {
+		return false, nil
+	}
+	noneSelector, err := strconv.ParseBool(noneSelectorStr)
+	if err != nil {
+		// When the user put an invalid noneSelector, we choose not to propagate this object to any child
+		// namespace to protect any object in the child namespaces to be overwritten
+		return true,
```

Comment:
That's a good point, but usually when you return an error, all other values are traditionally set to their zero values (`false`, for booleans) so I'd actually return false here, just like you return `nil` from the other two functions. It's confusing to Go developers to have a value returned from a failed function that they need to care about.

I know it makes the calling function a bit more complicated but I think it's worth it for consistency. I don't feel super-strongly about it though.

Reply:
Hm what I'm worrying here is that in reconcilers it will accidentally overwrite objects, in validators we don't care the return value anyways. What do you mean that "Go developers to have a value returned from a failed function that they need to care about"?

Reply:
Usually when a function returns (anything, error), and returns a non-nil error, the other return values are set to their zero-values (for a boolean, that's `false`) and ignored by the caller. So it's unusually to return `true, err` when `err` is non-nil; it's better to return `false, err` and then have the _caller_ decide what to do with the fact that `err` is non-nil. E.g., something like:

```
skip, err := shouldSkip()
if err != nil {
  log.Error(err, "Invalid value")
  skip = true
}
```

This is more consistent with how most other functions work in Go.